### PR TITLE
Add chapter 14 Logical Vectors notes

### PR DESCRIPTION
**Pull request by pevolution-ahmed**

### CODE REVIEW

File: 14-logical_vectors.Rmd [hunk 1]
```diff
@@ -1,1045 +1,229 @@
...
-str(yl)
-```
-
-- Lists can contain other lists!
-
-```{r 14-lists_within_lists, eval=FALSE}
-zl <- list(list(1, 2), list(3, 4))
-str(zl)
-```
+-   What are the logical summery functions we could leverage in R?
```
Comment (jonthegeek) [inline 14-logical_vectors.Rmd:13]:
```suggestion
-   What are the logical summary functions we could leverage in R?
```

File: 14-logical_vectors.Rmd [hunk 2]
```diff
@@ -1,1045 +1,229 @@
...
 
-### Visualising lists
+-   Conditional Transformations using `if_else()` and `case_when()` functions
 
-- Let's see a visual representation of lists. E.g., 
+## What is logical Vectors
```
Comment (jonthegeek) [inline 14-logical_vectors.Rmd:17]:
```suggestion
## What are Logical Vectors
```

File: 14-logical_vectors.Rmd [hunk 3]
```diff
@@ -1,1045 +1,229 @@
...
-```
-
-The error message mentions subsetting with zero, which returns no values:
-
-```{r 14-subset_vec_zero, eval=FALSE}
-x[0]
-```
-
-This is useful if we want to create unusual data structures to functions with. (Jon, please expound!)
+-   What are Vectors and Why we talking about Logical Vectors
```
Comment (jonthegeek) [inline 14-logical_vectors.Rmd:5]:
```suggestion
-   What are Vectors and Why are we talking about Logical Vectors
```

File: 14-logical_vectors.Rmd [hunk 4]
```diff
@@ -1,1045 +1,229 @@
...
+## How to use logical vectors & Boolean Algebra
 
-1. Lists have rounded corners. Atomic vectors have square corners.
+![](images/14_venn_diagrams.png)
 
-2. Children are drawn inside their parent, and have a slightly darker background to make it easier to see the hierarchy.
+-   Combining logical vectors using Boolean algebra
 
-3. The orientation of the children (i.e. rows or columns) isn’t important, a row or column orientation are picked to either save space or illustrate an important property in the example. ??
+    -   Venn Diagrams here shows the logical relations between logical vectors
```
Comment (jonthegeek) [inline 14-logical_vectors.Rmd:31]:
```suggestion
    -   Venn diagram here shows the logical relations between logical vectors
```

File: 14-logical_vectors.Rmd [hunk 5]
```diff
@@ -1,1045 +1,229 @@
...
 
-```{r 14-subsetting_lists, eval=FALSE}
-a <- list(a = 1:3, b = "a string", c = pi, d = list(-1, -5))
+```{r}
+library(tidyverse)
+library(nycflights13)
 ```
 
-1. `[` extracts a sub-list. And the result will always be a list.
+-   There is an intermediate execution step that form the logical vectors behind the scene when you use `filter()`
```
Comment (jonthegeek) [inline 14-logical_vectors.Rmd:42]:
```suggestion
-   There is an intermediate execution step that forma the logical vector behind the scenes when you use `filter()`
```

File: 14-logical_vectors.Rmd [hunk 6]
```diff
@@ -1,1045 +1,229 @@
...
-str(a[[1]])
+-   To separate this intermediate step in it's own variable we use `mutate`
 
-str(a[[4]])
+```{r}
+flights |> mutate( daytime = dep_time > 600 & dep_time < 2000, approx_ontime = abs(arr_delay) < 20, .keep = "used" )
 ```
 
-3. `$` is a shorthand to extract named elements of a list. It works similarly to `[[` except that you don’t need to use quotes.
+-   R doesn't round by default the float numeric value, but R console is rounding close number to integers
```
Comment (jonthegeek) [inline 14-logical_vectors.Rmd:54]:
```suggestion
-   R doesn't round the float numeric value by default, but R console is rounding close number to integers
```

**Review by jonthegeek — COMMENTED** [review]
I made some wording tweaks, which I'll apply myself in a moment. The youtube urls also changed, so I need to check those on my laptop to find out why.

**Comment by pevolution-ahmed** [comment]
> I made some wording tweaks, which I'll apply myself in a moment. The youtube urls also changed, so I need to check those on my laptop to find out why.
Good Catch! I'd probably have to go over everything again to make sure everything is in order.

**Review by jonthegeek — APPROVED** [review]
Thanks!

**Pull request merged**